spec/factories/ext_management_system.rb

```diff
@@ -28,6 +28,12 @@
         create_list :storage, evaluator.storage_count, :ext_management_system => ems
       end
     end
+
+    trait :with_invalid_authentication do
+      after(:create) do |x|
+        x.authentications << FactoryGirl.build(:authentication, :resource => x, :status => "invalid")
```

Comment:
Looking good! Last thing: It's good to reach for `build` first when you can, but one to many relationships may or may not work with initialized-but-not-persisted objects on them (as often, you're testing something where something else does a lookup on this authentications, for example), depending on what you're doing. As this is an optional trait that won't be used unless people explicitly want it, you likely want to use `FactoryGirl.create` here.

Reply:
I got it! `create` is really better in this case, thanks for pointing